### ISSUE TITLE
Can't reproduce your result

### ISSUE THREAD

**Issue by ad-0001** [zylo117/Yet-Another-EfficientDet-Pytorch#424]
When I try to evaluate the model on the repository efficientdet-d0.pth using comman `python coco_eval.py -p coco -c 0 -w efficientdet-d0.pth` I got the following result. How to get the result mAP: 33.1?

Loading and preparing results...
DONE (t=2.67s)
creating index...
index created!
BBox
Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=57.52s).
Accumulating evaluation results...
DONE (t=5.16s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.032
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.152
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.002
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.001
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.010
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.077
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.040
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.063
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.077
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.009
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.054
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.164

**Comment by zylo117** [comment]
that's not possible, I've just replicated 33.1 on d0.
Did you eval on coco val set? Or did you modify the code?

**Comment by ad-0001** [comment]
I think I made a mistake, I will try again now and share the result. Thank youuuu!

**Comment by ad-0001** [comment]
Thank you I could reproduce your results!

Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.332
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.511
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.349
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.121
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.381
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.521
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.277
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.419
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.449
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.183
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.524
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.652

**Comment by ad-0001** [comment]
How do I achieve this result using training from scratch?
Do I have to change learning rate or something or can use this command `python train.py -c 0 --batch_size 64 --optim sgd --lr 8e-2`?
How many epochs did it train to get exactly the same mAP?
I am training using the same command for around 80 epochs and classification loss is not stable still

**Comment by nebulaed** [comment]
You could use `python train.py -c 0 --batch_size x --optim adamw --lr 1e-4` to train from scratch. Batch_size is determined by your computer’s GPU memory. I used this setting to train for about 20 epochs, and the evaluation showed that it worked.

**Comment by WangTianYuan** [comment]
@nebulaed Hello, I have a question, `python train.py -c 0 --batch_size x --optim adamw --lr 1e-4` means training from scratch. But in the paper, Efficientdet-D0 use Efficientnet-b0 pre-trained weight, right? so why train from scratch?

**Comment by nebulaed** [comment]
> @nebulaed Hello, I have a question, `python train.py -c 0 --batch_size x --optim adamw --lr 1e-4` means training from scratch. But in the paper, Efficientdet-D0 use Efficientnet-b0 pre-trained weight, right? so why train from scratch?

From my perspective, the repository didn't utilize Efficientnet-b0 pre-trained weight, but built a backbone in backbone.py and initialized it in utils/utils.py. This is what the author calls "train from scratch". If not, I can only use the weights provided by the author.
